sequencer/src/l1_client.rs

```diff
@@ -129,6 +130,34 @@ impl L1Client {
             }
         }
     }
+    /// Get fee info for each `Deposit` ocurring between `prev`
+    /// and `new`. Returns `Vec<FeeInfo>`
+    pub async fn get_pending_deposits(
+        &self,
+        prev_finalized: Option<u64>,
+        new_finalized: u64,
+        address: Address,
+    ) -> Vec<FeeInfo> {
+        let prev = prev_finalized.unwrap_or(0);
+        // When called from `Header::new` during testing both will be
+        // `0`. BUG: https://github.com/foundry-rs/foundry/issues/7245
+        // We treat this as if there were no deposits.
+        if prev == new_finalized {
+            return vec![];
+        }
+
+        // query for deposit events, errors are handled as "no deposits".
+        let events =
+            contract_bindings::fee_contract::FeeContract::new(address, Arc::new(&self.provider))
+                .deposit_filter()
+                .from_block(prev)
```

Comment:
I think this should be from `prev + 1`, since deposits in block `prev` should already have been processed when `prev` was finalized. Unless `prev_finalized` was `None`, then we haven't processed anything so far. So we may need something like `let prev = prev_finalized.map(|prev| prev + 1).unwrap_or(0)`.

Reply:
fixed in d85c24bb94c26ad18d1fb202f110d67f6f90de3e